sysl2/sysl/exporter/type_exporter.go

```diff
@@ -0,0 +1,146 @@
+package exporter
+
+import (
+	"fmt"
+	"strings"
+
+	proto "github.com/anz-bank/sysl/src/proto"
+	"github.com/go-openapi/spec"
+	"github.com/sirupsen/logrus"
+)
+
+type TypeExporter struct {
+	primitiveTypesMap map[proto.Type_Primitive]map[string]string
+	log               *logrus.Logger
+}
+
+func makeTypeExporter(logger *logrus.Logger) *TypeExporter {
+	return &TypeExporter{
+		primitiveTypesMap: map[proto.Type_Primitive]map[string]string{
+			proto.Type_NO_Primitive: {"format": "", "type": "object"},
+			proto.Type_BOOL:         {"format": "", "type": "boolean"},
+			proto.Type_INT:          {"format": "integer", "type": "number"},
+			proto.Type_FLOAT:        {"format": "double", "type": "number"},
+			proto.Type_DECIMAL:      {"format": "double", "type": "number"},
+			proto.Type_STRING:       {"format": "string", "type": "string"},
+			proto.Type_BYTES:        {"format": "string", "type": "string"},
+			proto.Type_STRING_8:     {"format": "string", "type": "string"},
+			proto.Type_DATE:         {"format": "string", "type": "string"},
+			proto.Type_DATETIME:     {"format": "string", "type": "string"},
+			proto.Type_XML:          {"format": "string", "type": "string"},
+		},
+		log: logger,
+	}
+}
+
+func (v *TypeExporter) exportTypes(syslTypes map[string]*proto.Type, swaggerTypes spec.Definitions) error {
```

Comment:
could just return the spec.Definitions instead of having them passed in, otherwise func should be named "populateTypes" or just "populate"

Reply:
method name is exportTypes, not sure what you mean here?

Reply:
export would suggest the return value is the output, you're not exporting types heree, you're populating a passed in object